addons/event/models/event.py

```diff
@@ -331,6 +340,31 @@ def mail_attendees(self, template_id, force_send=False, filter_func=lambda self:
     def _is_event_registrable(self):
         return True
 
+    @api.multi
+    def _get_ics_file(self):
+        """ Returns iCalendar file for the event invitation.
+            :returns a dict of .ics file content for each event
+        """
+        result = {}
+        if not vobject:
+            return result
+
+        for event in self:
+            cal = vobject.iCalendar()
+            cal_event = cal.add('vevent')
+
+            if not event.date_begin or not event.date_end:
```

Comment:
Those 2 fields are required I think, so we don't need to check if they are empty.